sdk/python/tests/unit/infra/online_store/test_dynamodb_online_store.py

```diff
@@ -158,33 +163,123 @@ def test_dynamodb_table_dynamodb_resource():
+        entity_keys=entity_keys,
+    )
+    assert stored_items is not None
+    assert len(stored_items) == len(data)
+    assert [item[1] for item in stored_items] == list(features)
+
+
+@mock_dynamodb2
+def test_dynamodb_online_store_update(repo_config, dynamodb_online_store):
+    """Test DynamoDBOnlineStore update method."""
+    # create dummy table to keep
+    db_table_keep_name = f"{TABLE_NAME}_keep_update"
+    _create_test_table(PROJECT, db_table_keep_name, REGION)
+    # create dummy table to delete
+    db_table_delete_name = f"{TABLE_NAME}_delete_update"
+    _create_test_table(PROJECT, db_table_delete_name, REGION)
+
+    dynamodb_store = dynamodb_online_store
+    dynamodb_store.update(
+        config=repo_config,
+        tables_to_delete=[MockFeatureView(name=db_table_delete_name)],
+        tables_to_keep=[MockFeatureView(name=db_table_keep_name)],
+        entities_to_delete=None,
+        entities_to_keep=None,
+        partial=None,
+    )
+
+    # check only db_table_keep_name exists
+    dynamodb_client = dynamodb_store._get_dynamodb_client(REGION)
+    existing_tables = dynamodb_client.list_tables()
+    existing_tables = existing_tables.get("TableNames", None)
+
+    assert existing_tables is not None
+    assert len(existing_tables) == 1
+    assert existing_tables[0] == f"test_aws.{db_table_keep_name}"
+
+
+@mock_dynamodb2
+def test_dynamodb_online_store_teardown(repo_config, dynamodb_online_store):
+    """Test DynamoDBOnlineStore teardown method."""
+    db_table_delete_name_one = f"{TABLE_NAME}_delete_teardown_1"
+    db_table_delete_name_two = f"{TABLE_NAME}_delete_teardown_2"
+    _create_test_table(PROJECT, db_table_delete_name_one, REGION)
+    _create_test_table(PROJECT, db_table_delete_name_two, REGION)
+
+    dynamodb_store = dynamodb_online_store
+    dynamodb_store.teardown(
+        config=repo_config,
+        tables=[
+            MockFeatureView(name=db_table_delete_name_one),
+            MockFeatureView(name=db_table_delete_name_two),
+        ],
+        entities=None,
+    )
+
+    # Check tables non exist
+    dynamodb_client = dynamodb_store._get_dynamodb_client(REGION)
+    existing_tables = dynamodb_client.list_tables()
+    existing_tables = existing_tables.get("TableNames", None)
+
+    assert existing_tables is not None
+    assert len(existing_tables) == 0
+
+
+@mock_dynamodb2
+def test_dynamodb_online_store_write_batch_non_duplicates(
+    repo_config, dynamodb_online_store
+):
     """Test DynamoDBOnline Store deduplicate write batch request items."""
     dynamodb_tbl = f"{TABLE_NAME}_batch_non_duplicates"
     _create_test_table(PROJECT, dynamodb_tbl, REGION)
     data = _create_n_customer_test_samples()
     data_duplicate = deepcopy(data)
     dynamodb_resource = boto3.resource("dynamodb", region_name=REGION)
     table_instance = dynamodb_resource.Table(f"{PROJECT}.{dynamodb_tbl}")
-    dynamodb_store = DynamoDBOnlineStore()
+    dynamodb_store = dynamodb_online_store
```

Comment:
maybe just call this dynamodb_store to start? or reuse the variable instead of re-assigning.

Reply:
makes sense to just call the fixture variable `dynamodb_online_store`

Reply:
@adchia thanks for the review, let me know if sth else is needed.